FTL.h

```diff
@@ -70,7 +70,7 @@
 enum { DATABASE_WRITE_TIMER, EXIT_TIMER, GC_TIMER, LISTS_TIMER, REGEX_TIMER, ARP_TIMER, LAST_TIMER };
 enum { QUERIES, FORWARDED, CLIENTS, DOMAINS, OVERTIME, WILDCARD };
 enum { DNSSEC_UNSPECIFIED, DNSSEC_SECURE, DNSSEC_INSECURE, DNSSEC_BOGUS, DNSSEC_ABANDONED, DNSSEC_UNKNOWN };
-enum { QUERY_UNKNOWN, QUERY_GRAVITY, QUERY_FORWARDED, QUERY_CACHE, QUERY_WILDCARD, QUERY_BLACKLIST, QUERY_EXTERNAL_BLOCKED };
+enum { QUERY_UNKNOWN, QUERY_GRAVITY, QUERY_FORWARDED, QUERY_CACHE, QUERY_WILDCARD, QUERY_BLACKLIST, QUERY_EXTERNAL_BLOCKED_IP, QUERY_EXTERNAL_BLOCKED_NULL, QUERY_EXTERNAL_BLOCKED_NXRA };
```

Comment:
It would be useful to document the difference between the external blocked query types.

Reply:
Where should the documentation happen? It is hidden in the telnet API which will be replaced by the new API in the future. The difference between them is:

- `QUERY_EXTERNAL_BLOCKED_IP` = Detected an IP address that is known to correspond to a blocking page of an upstream provider
- `QUERY_EXTERNAL_BLOCKED_NULL` = Upstream server returned `0.0.0.0` as address which suggests they blocked it
- `QUERY_EXTERNAL_BLOCKED_NXRA` = Upstream server returned `NXDOMAIN` with `RA` bit set

Reply:
The documentation should be located next to the enum definition.

Reply:
Actually, it would be best documented on the documentation site:
https://docs.pi-hole.net/ftldns/database/#supported-status-types